app/src/utils/makeCancelable.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import CancelledPromiseError from '$shared/errors/CancelledPromiseError'
+
+export type Cancelable = {
+    promise: Promise<any>,
+    cancel: () => void,
+}
+
+/*
+ * NOTE: Took it from https://github.com/facebook/react/issues/5465#issuecomment-157888325 and
+ *       revamped it a little bit.
+ */
+
+const makeCancelable = (promise: Promise<any>) => {
```

Comment:
You could call this file `promise.js` and add a comment that the purpose is to make any promise cancelable. Just to add some context :)